### PR TITLE
Dt refactor

### PR DESCRIPTION
**Pull request by DavidTTran**
- I couldn't get fastjson_api to work for some reason, so I ended up hand rolling the hash, converting it to json, then passing it back to the application
- I'd like to not have so many `return nil if ...` but for the time being it's the quickest way to handle failed requests.
- We should probably have a sad path in the case that we do hit the API request limit during production/demo-ing

### CODE REVIEW

File: app/controllers/recipes_controller.rb [hunk 1]
```diff
@@ -1,18 +1,31 @@
...
 
 class RecipesController < Sinatra::Base
 
   get '/recipes/complexSearch' do
-    SpoonacularService.new.complex_search(params).to_json
-    # RecipeSerializer.new(recipes)
+    recipes = SpoonacularService.new.complex_search(params)
+
+    return nil if recipes[:results] == []
+    return nil if recipes.nil?
```
Comment (alex-latham) [inline app/controllers/recipes_controller.rb]: can combine with `return nil if recipes[:results].empty? || recipes.nil?`

File: app/services/spoonacular_service.rb [hunk 2]
```diff
@@ -24,14 +24,15 @@ def search_by_ids(recipe_ids)
...
+    search_params[:type] = "main course" if search_params[:type] == nil
     { apiKey: ENV['SPOONACULAR_KEY'],
       includeIngredients: search_params[:ingredients],
       maxReadyTime: search_params[:time],
       diet: search_params[:diet], type: search_params[:type],
       addRecipeInformation: true, fillIngredients: true, ignorePantry: true,
-      minCalories: 0, minCholesterol: 0, minFat: 0, minProtein: 0, minCarbs: 0,
-      minSugar: 0, minSodium: 0,
-      number: 12, sort: 'min-missing-ingredients'
+      addRecipeNutrition: true,
```
Comment (alex-latham) [inline app/services/spoonacular_service.rb:33]: Good one!

**Review by alex-latham — APPROVED** [review]
Great work.

File: app/serializers/recipe_serializer.rb [hunk 3]
```diff
@@ -1,8 +1,35 @@
 class RecipeSerializer
-  incliude FastJsonapi::ObjectSerializer
-  attributes :ingredients
+  def self.call(recipe)
```
Comment (mikez321) [inline app/serializers/recipe_serializer.rb:2]: This overall isn't too bad.... I mean, I didin't have to hand roll it, but I was expecting a HUGE hash here!  Nice work getting it done!

**Review by mikez321 — APPROVED** [review]
Well too bad that fastJSON didn't work out, but at least the hand-roll wasn't over 100 lines long like the actual API call!

**Pull request merged**